src/storage/store/column.cpp

```diff
@@ -35,12 +35,21 @@ struct ReadInternalIDValuesToVector {
         uint32_t posInVector, uint32_t numValuesToRead, const CompressionMetadata& metadata) {
         KU_ASSERT(resultVector->dataType.getPhysicalType() == PhysicalTypeID::INTERNAL_ID);
 
-        auto buffer = std::make_unique<offset_t[]>(numValuesToRead);
-        compressedReader(frame, pageCursor, reinterpret_cast<uint8_t*>(buffer.get()), 0,
+        std::unique_ptr<offset_t[]> bufferMgr;
+        offset_t fixedSizeBuffer[DEFAULT_VECTOR_CAPACITY];
+        offset_t* bufferPtr = nullptr;
+        if (numValuesToRead <= DEFAULT_VECTOR_CAPACITY) {
+            bufferPtr = fixedSizeBuffer;
+        } else {
+            bufferMgr = std::make_unique<offset_t[]>(numValuesToRead);
+            bufferPtr = bufferMgr.get();
```

Comment:
Should this case be handled or is it fine to replace it with `KU_ASSERT(numValuesToRead <= DEFAULT_VECTOR_CAPACITY)`?

Reply:
I think you can just assert it. I don't think value vectors can have a size other than `DEFAULT_VECTOR_CAPACITY` as the constructor always initializes the value buffer to that (I feel like it should have a better name).
There's a doc comment that seems to suggest that the capacity may also be 1, but I think that must be out of date.
>  The capacity of a ValueVector is either 1 (sequence) or DEFAULT_VECTOR_CAPACITY.

Reply:
I think the auxiliary buffer can resize it to something else but I don't think that's applicable here